cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -136,6 +137,32 @@ class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
       fs::FileLocator destination_locator) const override;
 
   std::shared_ptr<FileWriteOptions> DefaultWriteOptions() override;
+
+  /// \brief A getter function to retrie‰ve the dataset encryption configuration
+  std::shared_ptr<DatasetEncryptionConfiguration> GetDatasetEncryptionConfig() const {
+    return dataset_encryption_config_;
+  }
+  /// \brief A getter function to retrieve the dataset decryption configuration
+  std::shared_ptr<DatasetDecryptionConfiguration> GetDatasetDecryptionConfig() const {
+    return dataset_decryption_config_;
+  }
+  /// \brief A setter for DatasetEncryptionConfiguration
+  void SetDatasetEncryptionConfig(
+      std::shared_ptr<DatasetEncryptionConfiguration> dataset_encryption_config) {
+    dataset_encryption_config_ = dataset_encryption_config;
+  }
+  /// \brief A setter for DatasetDecryptionConfiguration
+  void SetDatasetDecryptionConfig(
+      std::shared_ptr<DatasetDecryptionConfiguration> dataset_decryption_config) {
+    dataset_decryption_config_ = dataset_decryption_config;
+  }
+
+ private:
+  // A configuration structure that provides per file encryption properties for a dataset
+  std::shared_ptr<DatasetEncryptionConfiguration> dataset_encryption_config_ = nullptr;
```

Comment:
You'll see in other header files we always use the `NULLPTR` macro:
```suggestion
  std::shared_ptr<DatasetEncryptionConfiguration> dataset_encryption_config_ = NULLPTR;
```